app/customer.py

```diff
@@ -0,0 +1,87 @@
+import math
+
+
+class Shop:
+    shops = []
+
+    def __init__(self, shop_dict: dict) -> None:
+        self.name = shop_dict["name"]
+        self.location = shop_dict["location"]
+        self.product = shop_dict["products"]

```

Comment:
use get() method instead of []
not `self.name = shop_dict["name"]`
but `self.name = shop_dict.get("name")`
change it in every line

Reply:
cool idea